src/test/java/com/ericsson/ei/controller/TestQueryControllerImpl.java

```diff
@@ -48,27 +54,35 @@
 @ContextConfiguration(classes = App.class, loader = SpringBootContextLoader.class, initializers = TestContextInitializer.class)
 @RunWith(SpringJUnit4ClassRunner.class)
 @SpringBootTest(classes = { App.class })
-public class TestQueryControllerImpl extends ControllerTestBaseClass {
+@AutoConfigureMockMvc
+public class TestQueryControllerImpl {
     private static final String inputPath = "src/test/resources/AggregatedObject.json";
     private static final String QUERY = "{\"criteria\" :{\"testCaseExecutions.testCase.verdict\":\"PASSED\", \"testCaseExecutions.testCase.id\":\"TC5\" }, \"options\" :{ \"id\": \"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43\"} }";
     private static String input;
 
     @MockBean
     private ProcessQueryParams unitUnderTest;
 
+    @Autowired
+    private MockMvc mockMvc;
+
     @Before
     public void setUp() {
         input = FileUtils.readFileAsString(new File(inputPath));
     }
 
     @Test
-    public void filterFormParamTest() throws Throwable {
+    public void filterFormParamTest() throws Exception {
+
         JSONArray inputObj = new JSONArray("[" + input + "]");
-        when(unitUnderTest.runQuery(any(JSONObject.class), any(JSONObject.class),
-                any(String.class))).thenReturn(inputObj);
+        when(unitUnderTest.runQuery(any(JSONObject.class), any(JSONObject.class), any(String.class)))
+                .thenReturn(inputObj);
 
-        MvcResult result = performMockMvcRequest("/query", QUERY);
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.post("/aggregated-objects/query").accept(MediaType.ALL)
```

Comment:
I see that we hardcode all entrypoints as text in several places.
I think we can create a EntryPointContants class, where all entrypoints in text is defined.
e.g. EntryPointConstants.AGGREGATED_OBJECTS , EntryPointConstants.SUBSCRIPTIONS,, and so. You can maybe start with the Aggregated-objects entry point in this PR.

Reply:
Although Tobias idea is great, it may be too large change for this story.

Reply:
I did it for the /aggregated-objects end point